### PR TITLE
feat: add an exception to naming convention

### PR DESCRIPTION
**Pull request by mad-it**
Discussion thread: https://ridedott.slack.com/archives/CDW5BKQJ1/p1611063181010800

### CODE REVIEW

**Comment by mad-it** [comment]
@viestat Can you help me with the tests failing? I extended the expiration but that didnt work 🤔

File: rules/typescript.js [hunk 1]
```diff
@@ -166,6 +166,15 @@ module.exports = {
...
       },
+      // Allow certain keywords that are used by engineering at Dott.
+      {
+        filter: {
+          match: true,
+          regex: '_version',
+        },
+        format: null,
+        selector: 'default',
+      },
```
Comment (viestat) [inline rules/typescript.js:177]: This should be accompanied by its own test.

In general terms, you need to add a passing fixture and a failing fixture.

**Review by viestat — CHANGES_REQUESTED** [review]
Tests for this configuration are missing.

I can led you a hand tomorrow if you need help. I will also take a look at why the test where the TODO was extended is failing.

**Comment by mad-it** [comment]
@viestat Added some tests with https://github.com/ridedott/eslint-config/pull/756/commits/ce66df6cebe6de904ab62eb714f991e671167716.
However I did notice that we have 2 of the same rules:  `array-func/prefer-flat-map` and `unicorn/prefer-flat-map`. Probably have to disable one of them. Can you help out with this?

**Review by viestat — APPROVED** [review]

**Comment by viestat** [comment]
> However I did notice that we have 2 of the same rules: `array-func/prefer-flat-map` and `unicorn/prefer-flat-map`. Probably have to disable one of them. Can you help out with this?

I can create a separate PR removing either one of them.

> Edit: Think we can use this PR to update all the deps as well? I fixed the `TODO` and this was the only thing blocking all the other PR's. That way we can group the release with 1 big release.

Im not sure what the benefit would be, however we would loose the relevance of the PR, if we would like to revert it, we would revert a lot of changes. I say we merge it and deal with the dependencies separately. Either way dependabot will try to update to the latest version.

**Pull request merged**